### ISSUE TITLE
`no-self-compare` does not handle complex expressions correctly

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#7677]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 7.2.0
* **npm Version:** 3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

(none)

**What did you do? Please include the actual source code causing the issue.**

```js
/* eslint no-self-compare: error */

foo.bar === foo.bar;
```

**What did you expect to happen?**

I expected ESLint to report an error for the self-comparison.

**What actually happened? Please include the actual, raw output from ESLint.**

No errors were reported.

I'm marking this as "evaluating" since I'm not sure whether it's intended behavior.
